### PR TITLE
python3Packages.pygeoip: drop

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

- #326513

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Sigmanificient** [comment]
@emilazy @natsukium Having trouble with `infinite recursion` (`sopel` -> `sopel-help` -> `sopel` -> ...)

Help would be welcome

**Comment by Sigmanificient** [comment]
@mog (sopel maintainer)

**Comment by natsukium** [comment]
Please remove `sopel-help` from `sopel`'s dependencies, since `sopel` doesn't directly depend on `sopel-help`.
https://github.com/search?q=repo%3Asopel-irc%2Fsopel+sopel_help&type=code

**Comment by Sigmanificient** [comment]
> Please remove `sopel-help` from `sopel`'s dependencies, since `sopel` doesn't directly depend on `sopel-help`. [github.com/search?q=repo%3Asopel-irc%2Fsopel+sopel_help&type=code](https://github.com/search?q=repo%3Asopel-irc%2Fsopel+sopel_help&type=code)

I added it because of:
```
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for sopel-8.0.0-py3-none-any.whl
  - sopel-help not installed
```

**Comment by emilazy** [comment]
This seems reasonable but it’d probably be best to split out the substantive `sopel` changes into another PR, as they don’t seem necessary to drop this package and I’m not entirely sure how we should handle the circular dependency here.

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 330509` at 5f4744f0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/natsukium/b3d1c26db6100b545a9ef87f5cdad822">python311Packages.sopel-help</a></li>
    <li><a href="https://gist.github.com/natsukium/4f3775314ddd7e572fafdb74e0b7560b">python312Packages.sopel-help</a></li>
  </ul>
</details>
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>nixpkgs-manual</li>
    <li>python311Packages.sopel</li>
    <li>python312Packages.sopel</li>
  </ul>
</details>

**Comment by natsukium** [comment]
I think the current hack is fine for the circular dependency, but sopel-help tests don't pass.

**Comment by emilazy** [comment]
I was thinking we’d probably want a standard `sopel` installation to include `sopel-help`, since that seems to be the intention of the upstream, but I don’t know how that would be best accomplished (a wrapper package?). In any case it should be handled in another PR, I think.

**Comment by natsukium** [comment]
We are working on separating build and runtime inputs for python packages in the future, and it will be resolved then. https://github.com/NixOS/nixpkgs/issues/272178

**Comment by Sigmanificient** [comment]
Just removed `sopel-help` since is it no longer require, will open pr if I get it to build

File: pkgs/development/python-modules/sopel/default.nix [hunk 1]
```diff
@@ -28,43 +29,48 @@ buildPythonPackage rec {
...
     pyenchant
-    pygeoip
     pytz
     sqlalchemy
     xmltodict
+    importlib-metadata
   ];
 
-  nativeCheckInputs = [ pytestCheckHook ];
+  pythonRemoveDeps = [ "sopel" ];
```
Comment (emilazy) [inline pkgs/development/python-modules/sopel/default.nix:51]: Should this be `"sopel-help"`?

**Review by emilazy — COMMENTED** [review]
@ofborg build python3Packages.sopel

File: pkgs/development/python-modules/sopel/default.nix [hunk 2]
```diff
@@ -28,43 +29,48 @@ buildPythonPackage rec {
...
-    pushd $TESTDIR
-  '';
+  nativeCheckInputs = [ pytestCheckHook ];
 
-  postCheck = ''
-    popd
-  '';
+  disabledTests = [
+    # requires network access
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/sopel/default.nix:59]:
```suggestion
```

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
@ofborg build python312Packages.sopel

**Pull request merged**